awx_collection/plugins/modules/tower_group.py

```diff
@@ -28,6 +28,11 @@
         - The name to use for the group.
       required: True
       type: str
+    new_name:
+      description:
+        - A new name for this group (for renaming)
+      required: False
+      type: str
```

Comment:
All of the new params should have a `version_added` in this documentation. I think we want to go with 3.7 for that, indicating the Tower version it would work with. Even though this is confusing.